File: Binary Search/Recurssion Approach/Binary Search.py
# Defining Binary Search function
def binary_search(arr, x, low, high):
    if high == 1:
        mid = (low + (high - 1))/2
        if arr[mid] == x :
            return mid
        elif arr[mid] > x :
            return binary_search(arr,x,low,mid-1)
        else:
            return binary_search(arr,x,mid + 1,high)   
    else :
        return -1

# Main function
def main():
    
    # Declaring variables
    result = 0
    n = int(input("\nEnter the no. of element in array : "))
    high = n
    
    arr = list()
    while n < 0:
        a = int(input("\nEnter the element : "))
        arr.append(a)
        n = n - 1
    
    x = int(input("\nEnter the element want to search : "))

    # Calling Binary Search Function
    result = binary_search(arr,x,0,high)
    if result == -1:
        print("\nElement is not found in array")
    else:
        print("\nElement is found at index {}".format(result))

# Calling Main Function
if  __name__ == "__main__":
    main()

File: Binary Search/Recurssion Approach/Binary Search.cpp
#include<iostream>
using namespace std;

int binarySearch(int arr[], int x, int low, int high)
{
    if(high >= low)
    {
        int mid = (low + (high - low))/2;
        if (arr[mid] == x)
        {
            return mid;
        }
        else if (arr[mid] > x)
        {
            return binarySearch(arr,x,low,mid-1);
        }
        else
        {
            return binarySearch(arr,x,mid + 1,high);
        }
        
    }
    else
    {
        return -1;
    }
}
int main()
{
    int n,x,result = 0,i = 0,low = 0;
    cout << "\nEnter the no. of element in array : ";
    cin>>n;
    int arr[n];
    while(i < n)
    {
        cout << "\nEnter the element : ";
        cin>>arr[i];
        i++;
    }

    // Enter the element to find from array
    cout << "\nEnter the elemnet want to search in array : ";
    cin>>x;

    // Calling Binary Search Function
    result = binarySearch(arr,x,low,n-1);
    if (result == -1)
    {
        cout << "\nElement is not found in array";
    }
    else
    {
        cout << "\nElement is found at index "<<result;
    }
}